core/quill.js

```diff
@@ -207,6 +213,7 @@ class Quill {
   }
 
   setContents(delta, source = Emitter.sources.API) {
+    this.toggleEmpty(this.getLength() <= 1);
```

Comment:
setContents should trigger a TEXT_CHANGE event so an explicit change to `ql-empty` should not be necessary